spark/src/main/java/org/apache/iceberg/spark/source/IcebergSource.java

```diff
@@ -189,10 +189,11 @@ private static void mergeIcebergHadoopConfs(
         .forEach(key -> baseConf.set(key.replaceFirst("hadoop.", ""), options.get(key)));
   }
 
-  private void validateWriteSchema(Schema tableSchema, Schema dsSchema, Boolean checkNullability) {
+  private void validateWriteSchema(
+          Schema tableSchema, Schema dsSchema, Boolean checkNullability, Boolean checkOrdering) {
     List<String> errors;
     if (checkNullability) {
-      errors = CheckCompatibility.writeCompatibilityErrors(tableSchema, dsSchema);
+      errors = CheckCompatibility.writeCompatibilityErrors(tableSchema, dsSchema, checkOrdering);
     } else {
       errors = CheckCompatibility.typeCompatibilityErrors(tableSchema, dsSchema);
```

Comment:
@ravichinoy checkOrdering needs to be passed here as well. You should be able to turn off both the nullability and ordering checks.